checkov/terraform/checks/terraform/terraform/StateLock.py

```diff
@@ -23,7 +23,11 @@ def scan_terraform_block_conf(self, conf: Dict[str, List[Any]]) -> CheckResult:
             return CheckResult.UNKNOWN
 
         s3_config = backend["s3"]
-        if ("use_lockfile" not in s3_config or not s3_config["use_lockfile"]) and "dynamodb_table" not in s3_config:
+        if isinstance(s3_config, list):
+            # this can happen for CDKTF output files
+            s3_config = s3_config[0]
+
+        if not s3_config.get("use_lockfile") and "dynamodb_table" not in s3_config:
```

Comment:
Couple of questions for my understanding: 
1. The user can use other backend in order to lock the state file, so why failing.
2. If he uses the dynamodb table for locking, don't we need to check if he actually created it?
Thank you

Reply:
1. if you check a few lines higher, then you can see the check is terminating, if there is no `s3` config. So, you wouldn't get a PASSED or FAILED result for any other backend config like `gcs`. 
2. how do you want to check, if someone created the needed DynamoDB table with static scanning? it is also not needed, because a TF run with an unknown DynamoDB table would fail anyway.